src/__tests__/ParseObject-test.js

```diff
@@ -869,12 +876,6 @@ describe('ParseObject', () => {
 
   it('can validate attributes', () => {
     const o = new ParseObject('Listing');
-    expect(
-      o.validate({
-        ACL: 'not an acl',
-      })
-    ).toEqual(new ParseError(ParseError.OTHER_CAUSE, 'ACL must be a Parse ACL.'));
```

Comment:
So now instead of throwing an error, setting an invalid ACL silently fails by setting the ACL to `null`?

Reply:
Yes, also users have the option to pass `ignoreValidation` option to `set`, in that case `getACL` would return `null`.

Reply:
So that's a change in behavior, right? Or can we say that throwing an error as it currently does is a bug?

Reply:
I would consider this to be a bug. I'm not entirely sure why its there. There are no integration tests in this SDK or the Server. Also this error message doesn't exist in any other SDK.

Reply:
I don't see how users would encounter this error if they follow the ACL documentation unless they set ACL without Parse.ACL like I'm doing in this PR. Thats probably the reason why there aren't any integration tests. If they some how do bypass the sdk internals with invalid ACL the server throws `Tried to create an ACL with an invalid permission type`.

Reply:
Do you mean that using the official methods of the SDK it's currently not possible to invoke `ParseError.OTHER_CAUSE, 'ACL must be a Parse ACL.'`?

Reply:
Correct, you can try to write a failing test yourself

Reply:
What about `o.setACL('invalid');`, wouldn't that throw the error?

Reply:
Surprisingly `o.setACL('invalid');` wouldn't throw an error it would return an error which wouldn't be seen by the developer.

Reply:
@mtrezza I have an idea that could work. Both `save` and `set` both validates the attributes. `save` calls `set` internally, removing the redundant validation should keep the error and allow for this feature.

Reply:
Yes, maybe that's a way forward. Just so that we avoid a behavior change that could break existing apps. But I agree that the current behavior looks odd as well.

What would be the ideal behavior a developer would want to see? I think `o.setACL('invalid');` should throw an error, right? At worst, a developer tries to set a ACL incorrectly and it fails silently, creating objects without ACL.

Reply:
Backbone support was removed in 2.0.0 I don't know how this got missed.